PSReadLine/ReadLine.vi.cs

```diff
@@ -593,7 +593,7 @@ public static void ViInsertWithDelete(ConsoleKeyInfo? key = null, object arg = n
             var isEOL = _singleton._current == _singleton._buffer.Length + ViEndOfLineFactor;
 
             DeleteChar(key, arg);
-            if(isEOL)
+            if(isEOL && _singleton._buffer.Length > 0)

```

Comment:
Could this addition just be part of the isEOL expression?  Placed first would allow it to short-circuit the rest of the expression.

Reply:
I'm fixing it in a more general way. Will update the PR.